### ISSUE TITLE
Pivot Table AddDateGrouping method is invalid

### ISSUE THREAD

**Issue by DukeJin-Dev** [EPPlusSoftware/EPPlus#820]
i create a pivot table, row filed tepe is date type, and add group by years and months, but is invalid , i don't known why.
![image](https://user-images.githubusercontent.com/85550799/225798514-d318f6b7-5d94-48bd-abc5-25b8f96e8e47.png)

![image](https://user-images.githubusercontent.com/85550799/225798922-3d8a25a8-e7c9-4055-ac8f-91a458b0206e.png)
  

i need it is group ,like this:
![image](https://user-images.githubusercontent.com/85550799/225798856-af74304f-965e-4a99-9601-070819e257fc.png)

**Comment by JanKallman** [comment]
You have a sample of date grouping here ... https://github.com/EPPlusSoftware/EPPlus.Sample.NetCore/blob/3c94b6c467e9891f6da280c018d8b8dc218c5311/18-PivotTables/PivotTablesSample.cs#L103
If something isn't working correctly with the date grouping please attach a unit test to reproduce the issue.

**Comment by DukeJin-Dev** [comment]
this question is appear when the rowField type is 'DateTime?' and the value is null, i don't known why, i try many version. 
this is test program: [ EPPlus PT Test Program](https://github.com/DukeJin-Dev/EPPlusPivotTableTest)

**Comment by JanKallman** [comment]
I'm unsure if i understand the problem here, but if you want the look of the grouping in the attached image, try this..
```
pt.SetCompact(true);
pt.RowFields[0].Outline = true;
pt.RowFields[1].Outline = true;
```

**Comment by DukeJin-Dev** [comment]
i try it but it is not work. i think the problem is 'the date group is not work when rowfield is datetime type and value is null'.

**Comment by JanKallman** [comment]
Yes, I can see the issue. As you did not have any null values in your sample, it was not obvious. I will have a look at it.

**Comment by JanKallman** [comment]
Fixed in EPPlus 8.0.4
